module2/dao/src/test/java/com/epam/esm/dao/GiftCertificateDaoTest.java

```diff
@@ -0,0 +1,86 @@
+package com.epam.esm.dao;
+
+import com.epam.esm.dao.entity.GiftCertificate;
+import org.hamcrest.collection.IsEmptyCollection;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.postgresql.util.PGInterval;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class GiftCertificateDaoTest {
+
+    private final GiftCertificateDao giftCertificateDao = mock(GiftCertificateDao.class);
```

Comment:
from requirements: 
Repository layer should be tested using integration tests with an in-memory embedded database (all operations with certificates).
But GiftCertificateDao is just mocked here.

Reply:
fixed